packages/flame/lib/src/components/mixins/has_hitboxes.dart

```diff
@@ -15,9 +15,14 @@ mixin HasHitboxes on PositionComponent {
 
   @override
   @mustCallSuper
-  void update(double dt) {
-    super.update(dt);
-    _validAabb = false;
+  void onMount() {
+    super.onMount();
+    void addCallback(PositionComponent c) {
+      c.transform.addListener(() => _validAabb = false);
+    }
+
+    ancestors().whereType<PositionComponent>().forEach(addCallback);
```

Comment:
How about adding a flag to `ancestors()` such that
```dart
if (includeSelf) {
  yield this;
}
```

Reply:
Good idea, it doesn't really follow the function name, but since the named variable makes it quite clear I changed to using that.